spec/index.md

```diff
@@ -5226,6 +5253,25 @@ verify_response(Q, R, Cert') ∧ lookup(["time"], Cert') = Found S.system_time /
 
 ```
 
+State after
+
+```html
+
+S with
+    query_stats[Q.receiver] = S.query_stats[Q.receiver] · {
+        timestamp = S.time[Q.receiver]
+        num_instructions = NumInstructions
+        num_request_payload_bytes = |Q.Arg|
+        num_response_payload_bytes =
+          if composite_query_helper(S, MAX_CYCLES_PER_QUERY, 0, Q.canister_id, Q.sender, Q.canister_id, Q.method_name, Q.arg) = (Reject (RejectCode, RejectMsg), _) then |RejectMsg|
+          else if composite_query_helper(S, MAX_CYCLES_PER_QUERY, 0, Q.canister_id, Q.sender, Q.canister_id, Q.method_name, Q.arg) = (Reply Res, _) then |Res|
+    }
+
+```
+
+where `NumInstructions` is the unspecified number of WebAssembly instructions executed during the evaluation of the query call,
+i.e., the value of `ic0.performance_counter(0)` at the end of the query method execution.
```

Comment:
```suggestion
where `NumInstructions` is the unspecified number of WebAssembly instructions executed during the evaluation of the query call,
i.e., the sum of values of `ic0.performance_counter(0)` at the end of all query and composite query method executions during the evaluation of the query call.
```
@Dfinity-skaestle Is my above suggestion correct?

Reply:
For collecting query stats during composite query execution, I think it makes most sense to track each canister individually (instead of grouping in the root canister). E.g. if a users queries canister A which in turn queries canister B, we would record two query calls, one on canister A and one on B, with the respective number of instructions and request / response sizes.